.github/workflows/deploy.yaml

```diff
@@ -28,27 +31,162 @@ jobs:
+            SHOULD_DEPLOY=false
+          fi
+
+          NEW_PUBLISHED_VERSION="$MAJOR.$MINOR.$PATCH"
+          echo "NEW_PUBLISHED_VERSION=$NEW_PUBLISHED_VERSION" >> $GITHUB_OUTPUT
+          echo "should_deploy=$SHOULD_DEPLOY" >> $GITHUB_OUTPUT
+
+      - name: Verify new published version
+        run: |
+          echo "NEW_PUBLISHED_VERSION: ${{ steps.compare_versions.outputs.new_published_version }}"
+        working-directory: '${{ matrix.value }}'
+      - name: Set version in version.json
+        if: ${{ env.IS_MAIN }} && steps.compare_versions.outputs.should_deploy == 'true'
+        run: |
+          NEW_PUBLISHED_VERSION=${{ steps.compare_versions.outputs.new_published_version }}
+          SCHEMA_VERSION=${{ steps.get_schema_version.outputs.schema_version }}
+          echo "{\"schemaVersion\": \"$SCHEMA_VERSION\", \"publishedVersion\": \"$NEW_PUBLISHED_VERSION\"}" > version.json
+        working-directory: '${{ matrix.value }}'
+      - name: Set Version in version.ts for version entity
+        if: ${{ env.IS_MAIN }} && steps.compare_versions.outputs.should_deploy == 'true'
+        run: npm run set-version
+        working-directory: '${{ matrix.value }}'
+      - name: Prepare Manifest if polygon-digital-carbon or carbonmark
+        if: (matrix.value == 'polygon-digital-carbon' || matrix.value == 'carbonmark') && steps.compare_versions.outputs.should_deploy == 'true'
+        run: npm run prepare-matic
+        working-directory: '${{ matrix.value }}'
+      - name: Commit all changes
+        if: steps.compare_versions.outputs.should_deploy == 'true'
+        run: |
+          set -x  # Enable debugging output
+          NEW_PUBLISHED_VERSION=${{ steps.compare_versions.outputs.new_published_version }}
+
+          git config user.email "deploy-action@carbonmark.com"
+          git config user.name "deploy-action"
+
+          git add version.json src/utils/version.ts
+
+          # Use git diff --cached --quiet to check for staged changes
+          if git diff --cached --quiet; then
+            echo "No changes to commit"
+          else
+            git commit -m "action: update to publishedVersion: $NEW_PUBLISHED_VERSION for ${{ matrix.value }}"
+            
+            if [[ "${GITHUB_REF}" == "refs/heads/"* ]]; then
+              BRANCH_NAME=${GITHUB_REF#refs/heads/}
+            elif [[ "${GITHUB_REF}" == "refs/pull/"* ]]; then
+              BRANCH_NAME=${GITHUB_HEAD_REF}
+            else
+              echo "Unsupported GITHUB_REF format: ${GITHUB_REF}"
+              exit 1
+            fi
+
+            echo "Pulling latest changes for branch: $BRANCH_NAME"
+            git pull --rebase origin $BRANCH_NAME
+
+            echo "Pushing to branch: $BRANCH_NAME"
+            git push origin $BRANCH_NAME
+          fi
+        working-directory: '${{ matrix.value }}'
+      - name: Tag the commit with the new version
+        if: ${{ env.IS_MAIN }} && steps.compare_versions.outputs.should_deploy == 'true'
```

Comment:
Here we agreed we should tag on staging instead, since commit hashes match on both staging and main